### PR TITLE
fix(components): [el-tree-select] fix lazy and multiple select node

### LINKED ISSUE #17232: [Component] [tree-select] el-tree-select开启lazy和multiple时的选中问题
Bug Type: **`Component`**

## Environment

- Vue Version: `3.5.0-alpha.2`
- Element Plus Version: `2.7.5`
- Browser / OS: ` Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36`
- Build Tool: `Vite`

### Related Component

- `el-tree-select`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGVsLXRyZWUtc2VsZWN0XG4gICAgdi1tb2RlbD1cInZhbHVlXCJcbiAgICBsYXp5XG4gICAgbXVsdGlwbGVcbiAgICBzaG93LWNoZWNrYm94XG4gICAgOnJlbmRlci1hZnRlci1leHBhbmQ9XCJ0cnVlXCJcbiAgICA6bG9hZD1cImxvYWRcIlxuICAgIDpwcm9wcz1cInByb3BzXCJcbiAgICBzdHlsZT1cIndpZHRoOiAyNDBweFwiXG4gIC8+XG4gIDxlbC1kaXZpZGVyIC8+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IGxhbmc9XCJ0c1wiIHNldHVwPlxuaW1wb3J0IHsgcmVmIH0gZnJvbSAndnVlJ1xuXG5jb25zdCB2YWx1ZSA9IHJlZihbXSlcbmNvbnN0IHByb3BzID0ge1xuICBsYWJlbDogJ2xhYmVsJyxcbiAgY2hpbGRyZW46ICdjaGlsZHJlbicsXG4gIGlzTGVhZjogJ2lzTGVhZicsXG59XG5cbmxldCBpZCA9IDBcblxuY29uc3QgbG9hZCA9IChub2RlLCByZXNvbHZlKSA9PiB7XG4gIGlmIChub2RlLmlzTGVhZikgcmV0dXJuIHJlc29sdmUoW10pXG5cbiAgc2V0VGltZW91dCgoKSA9PiB7XG4gICAgcmVzb2x2ZShbXG4gICAgICB7XG4gICAgICAgIHZhbHVlOiArK2lkLFxuICAgICAgICBsYWJlbDogYGxhenkgbG9hZCBub2RlJHtpZH1gLFxuICAgICAgfSxcbiAgICAgIHtcbiAgICAgICAgdmFsdWU6ICsraWQsXG4gICAgICAgIGxhYmVsOiBgbGF6eSBsb2FkIG5vZGUke2lkfWAsXG4gICAgICAgIGlzTGVhZjogdHJ1ZSxcbiAgICAgIH0sXG4gICAgXSlcbiAgfSwgNDAwKVxufVxuPC9zY3JpcHQ+XG4iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsInRzY29uZmlnLmpzb24iOiJ7XG4gIFwiY29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiBcIkVTTmV4dFwiLFxuICAgIFwianN4XCI6IFwicHJlc2VydmVcIixcbiAgICBcIm1vZHVsZVwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlUmVzb2x1dGlvblwiOiBcIkJ1bmRsZXJcIixcbiAgICBcInR5cGVzXCI6IFtcImVsZW1lbnQtcGx1cy9nbG9iYWwuZC50c1wiXSxcbiAgICBcImFsbG93SW1wb3J0aW5nVHNFeHRlbnNpb25zXCI6IHRydWUsXG4gICAgXCJhbGxvd0pzXCI6IHRydWUsXG4gICAgXCJjaGVja0pzXCI6IHRydWVcbiAgfSxcbiAgXCJ2dWVDb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IDMuM1xuICB9XG59XG4iLCJfbyI6e319)

### Steps to reproduce

开启懒加载和多选的el-tree-select，子节点(isLeaf不为true)未展开之前，无法选中
```vue
<template>
  <el-tree-select
    v-model="value"
    lazy
    multiple
    show-checkbox
    :render-after-expand="true"
    :load="load"
    :props="props"
    style="width: 240px"
  />
  <el-divider />
</template>

<script lang="ts" setup>
import { ref } from 'vue'

const value = ref([])
const props = {
  label: 'label',
  children: 'children',
  isLeaf: 'isLeaf',
}

let id = 0

const load = (node, resolve) => {
  if (node.isLeaf) return resolve([])

  setTimeout(() => {
    resolve([
      {
        value: ++id,
        label: `lazy load node${id}`,
      },
      {
        value: ++id,
        label: `lazy load node${id}`,
        isLeaf: true,
      },
    ])
  }, 400)
}
</script>

```

### What is Expected?

能够选中子节点（无论是否展开）

### What is actually happening?

无法选中

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by gaoxuan-haxibiao**
1.Fix the issue where non-leaf nodes cannot be selected when using lazy and multiple together.

2. Since the original code did not handle the selection of non-leaf nodes before expanding lazy-loaded child nodes, additional processing was added to link the selection of parent and child nodes.

3. Because the time required for lazyLoad after nodeExpand cannot be predicted, the triggering time of the node-expand event in the tree has been adjusted to occur within the callback of the expand function. This change only affects the timing when using lazyLoad.

BREAKING CHANGE :
no

closed closed Fixes #17232

Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by gaoxuan-haxibiao** [comment]
When selecting a non-leaf node, the display of `modelValue` was designed with reference to the `SHOW_CHILD` strategy from Ant Design  TreeSelect `showCheckedStrategy`. Has element-plus considered adding a `showCheckedStrategy` prop to the TreeSelect component?

File: packages/components/tree-select/src/tree.ts [hunk 1]
```diff
@@ -258,6 +265,34 @@ export const useTree = (
...
+
+          const cachedKeys = toValidArray(props.modelValue).filter(
+            (item) => !(item in dataMap) && !uncachedCheckedKeys.includes(item)
+          )
+
+          const childKeys = formatCheckedValues()
+          emit(UPDATE_MODEL_EVENT, cachedKeys.concat(childKeys))
+        }
+      })
+    },
```
Comment (yujinpan) [inline packages/components/tree-select/src/tree.ts:295]: Can the event be listened to externally after using this proxy method?

**Review by yujinpan — COMMENTED** [review]
👍

**Review by yujinpan — APPROVED** [review]
Thank you!

**Comment by gaoxuan-haxibiao** [comment]
@tolking  @btea  Can review and merge this pr?  Thank you!

**Review by tolking — APPROVED** [review]

**Review by btea — APPROVED** [review]

**Comment by btea** [comment]
Thanks!

**Pull request merged**